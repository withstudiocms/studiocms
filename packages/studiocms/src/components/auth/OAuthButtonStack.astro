---
import { useTranslations } from 'studiocms:i18n';
import { Divider } from 'studiocms:ui/components';
import OAuthButton from './OAuthButton.astro';
import { providerData, showOAuth } from './oAuthButtonProviders.js';

const lang = 'en-us';
const t = useTranslations(lang, '@studiocms/auth:oauth-stack');

const shouldShowOAuth = showOAuth && providerData.some(({ enabled }) => enabled);
---
{ shouldShowOAuth && ( 
	<Divider>
		<t-comp-key-oauth key="or-login-with">{t('or-login-with')}</t-comp-key-oauth>
	</Divider>
    <div class="button-stack">
	{
		providerData.map(({enabled, ...props}) => enabled && <OAuthButton {...props} />)
	}
    </div>
)}
<script>
	import { 
		$i18n,
		baseTranslation,
		makeTranslation,
	} from 'studiocms:i18n/client';

	const currentPage = '@studiocms/auth:oauth-stack';

	const i18n = $i18n(currentPage, baseTranslation[currentPage]);

	if (!customElements.get('t-comp-key-oauth')) {
		customElements.define('t-comp-key-oauth', makeTranslation(currentPage, i18n));
	}
</script>