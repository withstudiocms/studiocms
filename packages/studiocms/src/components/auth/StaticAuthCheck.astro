---
import type { UserSessionData } from 'studiocms:auth/lib/types';
import { getUserData } from 'studiocms:auth/lib/user';
import { StudioCMSRoutes } from 'studiocms:lib';

interface Props {
	userData: UserSessionData;
}
const { userData } = Astro.props as Props;

const isLoggedIn = userData.isLoggedIn;
---
<div 
  id="login-check" 
  style="display: none;" 
  data-is_logged_in={`${isLoggedIn}`}
  data-redirect_route={StudioCMSRoutes.mainLinks.dashboardIndex}
  ></div>

<script is:inline>
  const loginCheck = document.getElementById('login-check');

  if (loginCheck.dataset.is_logged_in === 'true') {
    window.location.href = loginCheck.dataset.redirect_route;
  }
</script>