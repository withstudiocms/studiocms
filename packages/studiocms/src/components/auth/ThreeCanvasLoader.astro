---
import type { SiteConfigCacheObject } from 'studiocms:sdk/types';

interface Props {
	config: SiteConfigCacheObject;
}

// Get the site config
const { config } = Astro.props as Props;

const { loginPageBackground, loginPageCustomImage } = config.data;
---
<div 
    style="display: none;" 
    id="auth-pages-config" 
    data-config_background={loginPageBackground} 
    data-config_custom_image={loginPageCustomImage}
    />

<script>
import "studiocms:auth/scripts/three";
</script>