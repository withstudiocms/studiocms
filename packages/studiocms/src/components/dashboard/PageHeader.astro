---
import { Button } from 'studiocms:ui/components/button';
import { Icon } from 'studiocms:ui/components/icon';

interface Props {
	title: string;
}

const { title } = Astro.props;
---
<header class="page-header">
  <div class="page-title-container">
    <Button color='primary' id='nav-open' class='mobile-btn'>
      <Icon name="heroicons:bars-3" height={24} width={24} />
    </Button>
    <h1 class="page-title">{title}</h1>
  </div>
  <div class="page-actions-container">
    <slot />
  </div>
</header>
<style>
.page-header {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  margin-bottom: 2rem;
    flex-wrap: wrap;
    gap: 1rem;

  & .page-title-container {
    display: flex;
    flex-direction: row;
    gap: 1rem;
    width: fit-content;

    & .page-title {
      display: block;
      width: fit-content;
      margin: 0;
    }
  }

  & .page-actions-container {
    display: flex;
    flex-direction: row;
    align-items: center;
    gap: 1rem;
  }
}
</style>
