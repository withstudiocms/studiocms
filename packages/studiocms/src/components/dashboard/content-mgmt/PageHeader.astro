---
import { type UiLanguageKeys, useTranslations } from 'studiocms:i18n';
import { SDKCoreJs, runSDK } from 'studiocms:sdk';
import { Button, Center, Group, Icon, Input, Modal } from 'studiocms:ui/components';

interface Props {
	lang: UiLanguageKeys;
	editPage?: boolean;
	createPage?: boolean;
	// title?: string;
	editFolder?: boolean;
	createFolder?: boolean;
}

const { editPage, createPage, createFolder, editFolder, lang } = Astro.props;

const t = useTranslations(lang, '@studiocms/dashboard:content-header');

const title = t('title');

let pageTitle = title;
let pageSlug: string | undefined = undefined;
let currentId = '';
let folderName = '';

const urlParams = Astro.url.searchParams;
const editId = urlParams.get('edit') || '';

if (editId) {
	const res = await runSDK(SDKCoreJs.GET.page.byId(editId));

	if (editPage) {
		pageTitle = `${res.data.title}`;
		pageSlug = res.data.slug;
		currentId = res.data.id;
	}
}

const folderId = urlParams.get('folder') || '';

if (folderId) {
	const res = await runSDK(SDKCoreJs.GET.folderList());

	const folder = res.data.find((folder) => folder.id === folderId);

	if (editFolder && folder) {
		pageTitle = `${folder.name}`;
		currentId = folder.id;
		folderName = folder.name;
	}
}

const currentMode = editPage
	? 'edit-page'
	: editFolder
		? 'edit-folder'
		: createFolder
			? 'create-folder'
			: createPage
				? 'create-page'
				: 'none';
---
<header class="page-header">
  <div class="page-title-container">
    <Button color='primary' id='nav-open' class='mobile-btn'>
      <Icon name="bars-3" height={24} width={24} slot="start-content" />
    </Button>
    <h1 class="page-title" data-link={Astro.locals.routeMap.sdk.pages}>
      {currentMode === 'none' ? (
        <t-content-header key="title">
          {pageTitle}
        </t-content-header>
      ) : currentMode === 'edit-folder' ? (
        <><t-content-header key="edit-folder-title">{t('edit-folder-title')}</t-content-header>: {folderName}</>
      ) : currentMode === 'edit-page' ? (
        <><t-content-header key="edit-page-title">{t('edit-page-title')}</t-content-header>: {pageTitle}</>
      ) : currentMode === 'create-folder' ? (
        <t-content-header key="create-folder-title">
          {t('create-folder-title')}
        </t-content-header>
      ) : (
        <t-content-header key="create-page-title">
          {t('create-page-title')}
        </t-content-header>
      )}
    </h1>
  </div>
  <div id="page-actions-container" class="page-actions-container">

    { editPage && ( 
      <Group>
        <Button id="delete-page-modal-trigger" color="danger" type="submit" size="sm" disabled={pageSlug === 'index'}>
            <Icon name="trash-20-solid" width={20} height={20} />
            <t-content-header key="delete-button">
              {t('delete-button')}
            </t-content-header>
        </Button>
        <Button id="edit-button" variant="solid" size="sm"  color="primary" type="submit" form="edit-page-form">
          <Icon slot="start-content" name="check-20-solid" width={20} height={20} />
          <t-content-header key="save-button">
            {t('save-button')}
          </t-content-header>
        </Button>
      </Group>
      <Modal 
        id='delete-page-modal'
        isForm
        cancelButton={{ label: 'Cancel', color: 'default' }}
        actionButton={{ label: 'Confirm', color: 'danger' }}
        >
        <h2 slot='header'>
          <t-content-header key="delete-modal-header">
            {t('delete-modal-header')}
          </t-content-header>
        </h2>
        <Center>
          <div class="modal-body">
            <input type="hidden" name="page-id" value={currentId} />
            <input type="hidden" name="page-slug" value={pageSlug} />
            <input type="hidden" name="action-route" value={Astro.locals.routeMap.endpointLinks.content.page} />
                    
            <span><t-content-header key="delete-modal-desc-1">
            {t('delete-modal-desc-1')}
          </t-content-header> <code>{pageSlug}</code> <t-content-header key="delete-modal-desc-2">
            {t('delete-modal-desc-2')}
          </t-content-header></span>
           
            <Input name="slug-confirm" placeholder={`${pageSlug?.slice(0, pageSlug.length-2)}...`} isRequired />
            
            <span style="color: hsl(var(--danger-base))"><t-content-header key="delete-modal-warning">
            {t('delete-modal-warning')}
          </t-content-header></span>
          </div>
        </Center>
      </Modal>
    )}

    { editFolder && ( 
      <Group>
        <Button id="delete-folder-modal-trigger" color="danger" type="submit" size="sm">
          <Icon name="trash-20-solid" width={20} height={20} />
          <t-content-header key="delete-button">
            {t('delete-button')}
          </t-content-header>
        </Button>
        <Button id="edit-button" variant="solid" size="sm"  color="primary" type="submit" form="edit-folder-form">
          <Icon slot="start-content" name="check-20-solid" width={20} height={20} />
          <t-content-header key="save-button">
            {t('save-button')}
          </t-content-header>
        </Button>
      </Group>
      <Modal
        id="delete-folder-modal"
        isForm
        cancelButton={{ label: 'Cancel', color: 'default' }}
        actionButton={{ label: 'Delete', color: 'danger' }}
        >
        <h2 slot="header">
          <t-content-header key="delete-modal-header">
            {t('delete-modal-header')}
          </t-content-header>
        </h2>
        <Center>
          <div class="modal-body">
            <input type="hidden" name="folder-id" value={currentId} />
            <input type="hidden" name="folder-name" value={folderName} />
            <input type="hidden" name="action-route" value={Astro.locals.routeMap.endpointLinks.content.folder} />

            <span><t-content-header key="delete-folder-modal-desc-1">
            {t('delete-folder-modal-desc-1')}
          </t-content-header> <code>{folderName}</code> <t-content-header key="delete-folder-modal-desc-2">
            {t('delete-folder-modal-desc-2')}
          </t-content-header></span>

            <Input name="confirm-folder-name" placeholder={`${folderName.slice(0, folderName.length-2)}...`} isRequired />
          
            <span style="color: hsl(var(--danger-base))"><t-content-header key="delete-modal-warning">
            {t('delete-modal-warning')}
          </t-content-header></span>
          </div>
        </Center>
      </Modal>
    )}

    { createFolder && ( 
      <Button id="edit-button" variant="solid" size="sm"  color="primary" type="submit" form="create-folder-form">
        <Icon slot="start-content" name="check-20-solid" width={20} height={20} />
          <t-content-header key="create-button">
            {t('create-button')}
          </t-content-header>
      </Button>
    )}

    {createPage && ( 
      <Group>
      <Button id="draft-button" variant="outlined" size="sm" color="primary" type="submit" form="page-create-form">
        <Icon slot="start-content" name="document-text-20-solid" width={20} height={20} />
          <t-content-header key="draft-button">
            {t('draft-button')}
          </t-content-header>
      </Button>
      <Button id="publish-button" variant="solid" size="sm"  color="primary" type="submit" form="page-create-form">
        <Icon slot="start-content" name="check-20-solid" width={20} height={20} />
          <t-content-header key="publish-button">
            {t('publish-button')}
          </t-content-header>
      </Button>
      </Group>
    )}
  </div>
</header>

<div
  id="current-mode-selector"
  style="display: none;"
  data-mode={currentMode}
></div>

<script>
    import { 
        $i18n,
        baseTranslation,
        makeTranslation,
    } from 'studiocms:i18n/client';

    const currentPage = '@studiocms/dashboard:content-header';

    const i18n = $i18n(currentPage, baseTranslation[currentPage]);
    
    if (!customElements.get('t-content-header')) {
        customElements.define('t-content-header', makeTranslation(currentPage, i18n));
    }
</script>

<script>
    import { ModalHelper } from 'studiocms:ui/components';
    import { toast } from "studiocms:ui/components";

    const mode = (document.getElementById('current-mode-selector') as HTMLDivElement).dataset.mode;

    switch (mode) {
      case "edit-page": {
        const deletePageModal = new ModalHelper('delete-page-modal', 'delete-page-modal-trigger');

        deletePageModal.registerConfirmCallback(async (formData) => {
            if (!formData) {
              toast({
                  title: 'Error',
                  description: 'Something went wrong',
                  type: 'danger'
              })
              return;
            }

            const slugConfirm = formData?.get('slug-confirm')?.toString();
            const action = formData.get('action-route') as string;

            const data = {
              id: formData?.get('page-id')?.toString(),
              slug: formData?.get('page-slug')?.toString(),
            }

            if (!slugConfirm) {
              toast({
                  title: 'Error',
                  description: 'Something went wrong',
                  type: 'danger'
              })
              return;
            }

            if (!data.id || !data.slug) {
              toast({
                  title: 'Error',
                  description: 'Something went wrong',
                  type: 'danger'
              })
              return;
            }

            if (slugConfirm !== data.slug) {
              toast({
                  title: 'Error',
                  description: 'Slug does not match',
                  type: 'danger'
              })
              return;
            }

            const response = await fetch(action, {
              method: 'DELETE',
              headers: {
                  'Content-Type': 'application/json',
              },
              body: JSON.stringify(data),
            });

            const res = await response.json();

            if (response.status !== 200) {
              toast({
                  title: 'Error',
                  description: res.error,
                  type: 'danger'
              })
              return;
            }

            if (response.status === 200) {
              toast({
                  title: 'Success',
                  description: res.message,
                  type: 'success'
              })
              return;
            }
        })
        break;
      }
      case "edit-folder": {
        const deleteFolderModal = new ModalHelper('delete-folder-modal', 'delete-folder-modal-trigger');

        deleteFolderModal.registerConfirmCallback(async (formData) => {
            if (!formData) {
              toast({
                  title: 'Error',
                  description: 'Something went wrong',
                  type: 'danger'
              })
              return;
            }

            type JsonData = {
              id: string;
              folderName: string;
              parentFolder: string | null;
            };

            const folderName = formData.get('folder-name') as string;
            const id = formData.get('folder-id') as string;
            const confirmFolderName = formData?.get('confirm-folder-name');
            const action = formData.get('action-route') as string;

            if (!folderName || !confirmFolderName) {
              toast({
                  title: 'Error',
                  description: 'Something went wrong',
                  type: 'danger'
              })
              return;
            }

            if (folderName !== confirmFolderName) {
              toast({
                  title: 'Error',
                  description: 'Folder name does not match',
                  type: 'danger'
              })
              return;
            }

            const data: JsonData = {
              id,
              folderName: folderName,
              parentFolder: null,
            };

            const response = await fetch(action, {
              method: 'DELETE',
              headers: {
                  'Content-Type': 'application/json',
              },
              body: JSON.stringify(data),
            });

            const res = await response.json();

            if (response.status !== 200) {
              toast({
                  title: 'Error',
                  description: res.error,
                  type: 'danger'
              })
              return;
            }

            if (response.status === 200) {
              toast({
                  title: 'Success',
                  description: res.message,
                  type: 'success'
              })
              return;
            }
        })
        break;
      }
      case "create-folder":
        break;
      case "create-page":
        break;
      default:
        break;
    }
</script>

<style>
.page-header {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  flex-wrap: wrap;
  gap: 1rem;
}

.page-title-container {
  display: flex;
  flex-direction: row;
  gap: 1rem;
  width: fit-content;
}

.page-title {
  display: block;
  width: fit-content;
  margin: 0;
}

.page-actions-container {
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: 1rem;
}

.modal-body {
    display: flex;
    flex-direction: column;
    gap: 1rem;

    span {
        font-size: .875rem;
    }

    code {
        background-color: hsl(var(--background-step-3));
        padding: .25rem;
        border-radius: 4px;
    }
}
</style>
