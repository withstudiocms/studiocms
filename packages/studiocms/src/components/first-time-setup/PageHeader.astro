---
import { Badge } from 'studiocms:ui/components';
import type { AvailableIcons } from 'studiocms:ui/icons';

interface Props {
	title: string;
	badge?: {
		label: string;
		icon?: AvailableIcons;
	};
}

const { title, badge } = Astro.props;
---
<header class="page-header">
  <div class="page-title-container">
    <h1 class="page-title">{title}</h1>
    { badge && <Badge label={badge.label} size="lg" icon={badge.icon} /> }
  </div>
  <div class="page-actions-container">
    <slot />
  </div>
</header>
<style>
.page-header {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 1rem;
  width: 100%;
  margin-left: 20%;

  & .page-title-container {
    display: flex;
    flex-direction: row;
    align-items: center;
    gap: .5rem;
    width: fit-content;

    & .page-title {
      display: flex;
      margin: 0;
      font-size: 3rem;
    }
  }

  & .page-actions-container {
    display: flex;
    flex-direction: row;
    align-items: center;
    gap: 1rem;
  }
}

@media screen and (max-width: 768px) {
  .page-header {
    margin-left: 0;
  }
}
</style>
