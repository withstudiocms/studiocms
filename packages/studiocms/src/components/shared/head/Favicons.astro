---
import { FAVICON_ASSETS } from '../../../consts.js';
---
<link rel="icon" href={FAVICON_ASSETS.svg} type="image/svg+xml" />
<link rel="icon" href={FAVICON_ASSETS.png.light} type="image/png" media="(prefers-color-scheme: dark)" />
<link rel="icon" href={FAVICON_ASSETS.png.dark} type="image/png" media="(prefers-color-scheme: light)" />