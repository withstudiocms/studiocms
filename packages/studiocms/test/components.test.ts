/// <reference types="astro/client" />
import { describe, expect } from 'vitest';
import FallbackCanvas from '../src/components/auth/FallbackCanvas.astro';
import OAuthButton from '../src/components/auth/OAuthButton.astro';
import OAuthButtonStack from '../src/components/auth/OAuthButtonStack.astro';
import StaticAuthCheck from '../src/components/auth/StaticAuthCheck.astro';
import StudioCMSLogoSVG from '../src/components/auth/StudioCMSLogoSVG.astro';
import ThreeCanvasLoader from '../src/components/auth/ThreeCanvasLoader.astro';
import PageHeader from '../src/components/first-time-setup/PageHeader.astro';
import Code from '../src/components/shared/Code.astro';
import Favicons from '../src/components/shared/head/Favicons.astro';
import Global from '../src/components/shared/head/Global.astro';
import TitleTags from '../src/components/shared/head/TitleTags.astro';
import SSRUser from '../src/components/shared/SSRUser.astro';
import ThemeManager from '../src/components/shared/ThemeManager.astro';
import { test } from './fixtures/AstroContainer';

describe('Components Container tests', () => {
	describe('Shared components', () => {
		describe('TitleTags', () => {
			test('render component', async ({ renderComponent }) => {
				const result = await renderComponent(TitleTags, 'TitleTags', {
					props: { title: 'Test Title', description: 'Test Description' },
				});
				expect(result).toMatchSnapshot();
			});
		});

		describe('ThemeManager component', () => {
			test('render component', async ({ renderComponent }) => {
				const result = await renderComponent(ThemeManager, 'ThemeManager');
				expect(result).toMatchSnapshot();
			});
		});

		describe('SSRUser component', () => {
			test('render component', async ({ renderComponent }) => {
				const result = await renderComponent(SSRUser, 'SSRUser', {
					props: { name: 'mock', description: 'mock-admin' },
				});
				expect(result).toMatchSnapshot();
			});
		});

		describe('Global component', () => {
			test('render component', async ({ renderComponent }) => {
				const result = await renderComponent(Global, 'Global');
				expect(result).toMatchSnapshot();
			});
		});

		describe('Favicons component', () => {
			test('render component', async ({ renderComponent }) => {
				const result = await renderComponent(Favicons, 'Favicons');
				expect(result).toMatchSnapshot();
			});
		});

		describe('Code component', () => {
			test('render component', async ({ renderComponent }) => {
				const result = await renderComponent(Code, 'Code', {
					props: { code: 'export const hello = "hello world!";', __test_mode: true },
				});
				expect(result).toMatchSnapshot();
			});
		});
	});

	describe('First Time Setup components', () => {
		describe('PageHeader component', () => {
			test('render component', async ({ renderComponent }) => {
				const result = await renderComponent(PageHeader, 'PageHeader', {
					props: { title: 'Test Page' },
				});
				expect(result).toMatchSnapshot();
			});
			test('render component with badge', async ({ renderComponent }) => {
				const result = await renderComponent(PageHeader, 'PageHeader', {
					props: { title: 'Test Page', badge: { label: 'New' } },
				});
				expect(result).toMatchSnapshot();
			});
			test('render component with badge and icon', async ({ renderComponent }) => {
				const result = await renderComponent(PageHeader, 'PageHeader', {
					props: { title: 'Test Page', badge: { label: 'New', icon: 'heroicons:academic-cap' } },
				});
				expect(result).toMatchSnapshot();
			});
		});
	});

	describe('Auth components', () => {
		describe('ThreeCanvasLoader component', () => {
			test('render component', async ({ renderComponent }) => {
				const result = await renderComponent(ThreeCanvasLoader, 'ThreeCanvasLoader');
				expect(result).toMatchSnapshot();
			});
		});

		describe('StudioCMSLogoSVG component', () => {
			test('render component', async ({ renderComponent }) => {
				const result = await renderComponent(StudioCMSLogoSVG, 'StudioCMSLogoSVG');
				expect(result).toMatchSnapshot();
			});
		});

		describe('StaticAuthCheck component', () => {
			test('render component no props', async ({ renderComponent }) => {
				const result = await renderComponent(StaticAuthCheck, 'StaticAuthCheck');
				expect(result).toMatchSnapshot();
			});
			test('render component with props', async ({ renderComponent }) => {
				const result = await renderComponent(StaticAuthCheck, 'StaticAuthCheck', {
					props: { userData: { isLoggedIn: true } },
				});
				expect(result).toMatchSnapshot();
			});
		});

		describe('OAuthButtonStack component', () => {
			test('render component', async ({ renderComponent }) => {
				const result = await renderComponent(OAuthButtonStack, 'OAuthButtonStack');
				expect(result).toMatchSnapshot();
			});
		});

		describe('OAuthButton component', () => {
			test('render component', async ({ renderComponent }) => {
				const result = await renderComponent(OAuthButton, 'OAuthButton', {
					props: {
						label: 'test auth',
						href: '/test/endpoint',
						image: '<img src="/fake/image.jpg">',
					},
				});
				expect(result).toMatchSnapshot();
			});
		});

		describe('FallbackCanvas component', () => {
			test('render component', async ({ renderComponent }) => {
				const result = await renderComponent(FallbackCanvas, 'FallbackCanvas');
				expect(result).toMatchSnapshot();
			});
		});
	});
});
