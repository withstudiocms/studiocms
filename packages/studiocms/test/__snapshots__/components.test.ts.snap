// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components Container tests > Auth components > FallbackCanvas component > render component 1`] = `"<div class="fallback-container" id="fallback-config" data-config="{&#34;loginPageBackground&#34;:&#34;studiocms-curves&#34;,&#34;loginPageCustomImage&#34;:null}"> <svg class="static-logo" width="755" height="792" viewBox="0 0 755 792" fill="none" xmlns="http://www.w3.org/2000/svg"> <rect x="295" width="460" height="466" rx="32" fill="currentColor"></rect> <path d="M272 434V166H180C162.327 166 148 180.327 148 198V597C148 614.673 162.327 629 180 629H577.5C595.173 629 609.5 614.673 609.5 597V490H328C297.072 490 272 464.928 272 434Z" fill="currentColor"></path> <path d="M124 597V329H32C14.3269 329 0 343.327 0 361V760C0 777.673 14.3269 792 32 792H429.5C447.173 792 461.5 777.673 461.5 760V653H180C149.072 653 124 627.928 124 597Z" fill="currentColor"></path> </svg>  <img src="/mock/path/FallbackCanvas.astro?href=%2Fmock%2Fpath%2Fimage.webp&#38;w=1920&#38;h=1080&#38;q=100&#38;f=webp" data-image-component="true" alt="A fallback image displaying because the interactive wallpaper couldn't be loaded." loading="lazy" decoding="async" fetchpriority="auto" width="1920" height="1080" class="fallback-image"> <script type="module" src="/mock/path/FallbackCanvas.astro?astro&type=script&index=0&lang.ts"></script> </div>"`;

exports[`Components Container tests > Auth components > OAuthButton component > render component 1`] = `"<a tabindex="0" as="a" href="/test/endpoint" class="sui-button full lg default solid">   <img src="/fake/image.jpg"> <span>test auth</span>   </a> "`;

exports[`Components Container tests > Auth components > OAuthButtonStack component > render component 1`] = `
"<div class="sui-divider-container"> <hr class="sui-divider-line"> <div class="sui-divider-content" style="background-color: var(--background-base); padding: .25rem .5rem;"> <t-comp-oauth key="or-login-with">or log in using</t-comp-oauth> </div> </div>
    <div class="button-stack"><a tabindex="0" as="a" href="/studiocms_api/auth/github" class="sui-button full lg default solid">   github.png <span>GitHub</span>   </a> </div><script type="module" src="/mock/path/OAuthButtonStack.astro?astro&type=script&index=0&lang.ts"></script>"
`;

exports[`Components Container tests > Auth components > StaticAuthCheck component > render component no props 1`] = `
"<div id="login-check" style="display: none;" data-is_logged_in="false" data-redirect_route="/dashboard"></div> <script>
  const loginCheck = document.getElementById('login-check');

  if (loginCheck.dataset.is_logged_in === 'true') {
    window.location.href = loginCheck.dataset.redirect_route;
  }
</script>"
`;

exports[`Components Container tests > Auth components > StaticAuthCheck component > render component with props 1`] = `
"<div id="login-check" style="display: none;" data-is_logged_in="true" data-redirect_route="/dashboard"></div> <script>
  const loginCheck = document.getElementById('login-check');

  if (loginCheck.dataset.is_logged_in === 'true') {
    window.location.href = loginCheck.dataset.redirect_route;
  }
</script>"
`;

exports[`Components Container tests > Auth components > StudioCMSLogoSVG component > render component 1`] = `"<svg width="755" height="792" viewBox="0 0 755 792" fill="none" xmlns="http://www.w3.org/2000/svg"> <rect x="295" width="460" height="466" rx="32" fill="currentColor"></rect> <path d="M272 434V166H180C162.327 166 148 180.327 148 198V597C148 614.673 162.327 629 180 629H577.5C595.173 629 609.5 614.673 609.5 597V490H328C297.072 490 272 464.928 272 434Z" fill="currentColor"></path> <path d="M124 597V329H32C14.3269 329 0 343.327 0 361V760C0 777.673 14.3269 792 32 792H429.5C447.173 792 461.5 777.673 461.5 760V653H180C149.072 653 124 627.928 124 597Z" fill="currentColor"></path> </svg>"`;

exports[`Components Container tests > Auth components > ThreeCanvasLoader component > render component 1`] = `"<div style="display: none;" id="auth-pages-config" data-config_background="studiocms-curves"></div> <script type="module" src="/mock/path/ThreeCanvasLoader.astro?astro&type=script&index=0&lang.ts"></script>"`;

exports[`Components Container tests > First Time Setup components > PageHeader component > render component 1`] = `"<header class="page-header"> <div class="page-title-container"> <h1 class="page-title">Test Page</h1>  </div> <div class="page-actions-container">  </div> </header> "`;

exports[`Components Container tests > First Time Setup components > PageHeader component > render component with badge 1`] = `"<header class="page-header"> <div class="page-title-container"> <h1 class="page-title">Test Page</h1> <span class="sui-badge primary lg outlined full">  New  </span> </div> <div class="page-actions-container">  </div> </header> "`;

exports[`Components Container tests > First Time Setup components > PageHeader component > render component with badge and icon 1`] = `"<header class="page-header"> <div class="page-title-container"> <h1 class="page-title">Test Page</h1> <span class="sui-badge primary lg outlined full"> <svg style="min-width: 24px" xmlns="http://www.w3.org/2000/svg" prefix="heroicons" height="24" width="24" viewbox="0 0 24 24"><path fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5" d="M4.26 10.147a60 60 0 0 0-.491 6.347A48.6 48.6 0 0 1 12 20.904a48.6 48.6 0 0 1 8.232-4.41a61 61 0 0 0-.491-6.347m-15.482 0a51 51 0 0 0-2.658-.813A60 60 0 0 1 12 3.493a60 60 0 0 1 10.399 5.84q-1.345.372-2.658.814m-15.482 0A51 51 0 0 1 12 13.489a50.7 50.7 0 0 1 7.74-3.342M6.75 15a.75.75 0 1 0 0-1.5a.75.75 0 0 0 0 1.5m0 0v-3.675A55 55 0 0 1 12 8.443m-7.007 11.55A5.98 5.98 0 0 0 6.75 15.75v-1.5"/></svg> New  </span> </div> <div class="page-actions-container">  </div> </header> "`;

exports[`Components Container tests > Shared components > Code component > render component 1`] = `
"<pre class="code-container scrollbar">  <code id="test-id" class="custom-code"><span class="line">export const hello = &quot;hello world!&quot;;</span></code>
</pre> <script>(function(){const id = "test-id";

	const codesWithDiff = document.getElementById(id);

	if (!codesWithDiff) return;

	const code = codesWithDiff.classList.contains("diff");

	if (!code) return;

	const linesInserted = codesWithDiff.getAttribute("data-lines-inserted");
	const linesDeleted = codesWithDiff.getAttribute("data-lines-deleted");

	if (!linesInserted && !linesDeleted) return;

	const parsedInsertedLines =
		linesInserted?.split(",").map((line) => parseInt(line, 10)) || [];
	const parsedDeletedLines =
		linesDeleted?.split(",").map((line) => parseInt(line, 10)) || [];

	const codeLines = codesWithDiff.querySelectorAll(".line");

	codeLines.forEach((line, index) => {
		if (parsedInsertedLines.includes(index + 1)) {
			line.classList.add("diff", "ins");
		}

		if (parsedDeletedLines.includes(index + 1)) {
			line.classList.add("diff", "del");
		}
	});
})();</script>"
`;

exports[`Components Container tests > Shared components > Favicons component > render component 1`] = `"<link rel="icon" href="https://cdn.studiocms.dev/favicon.svg" type="image/svg+xml"><link rel="icon" href="https://cdn.studiocms.dev/favicon-light.png" type="image/png" media="(prefers-color-scheme: dark)"><link rel="icon" href="https://cdn.studiocms.dev/favicon-dark.png" type="image/png" media="(prefers-color-scheme: light)">"`;

exports[`Components Container tests > Shared components > Global component > render component 1`] = `"<meta charset="utf-8"><meta name="viewport" content="width=device-width,initial-scale=1">"`;

exports[`Components Container tests > Shared components > SSRUser component > render component 1`] = `"<div class="sui-user-container"> <div class="sui-avatar-container"> <svg style="min-width: 24px" xmlns="http://www.w3.org/2000/svg" prefix="heroicons" role="img" aria-label="Placeholder avatar for mock" height="24" width="24" viewbox="0 0 24 24"><path fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5" d="M15.75 6a3.75 3.75 0 1 1-7.5 0a3.75 3.75 0 0 1 7.5 0M4.501 20.118a7.5 7.5 0 0 1 14.998 0A17.9 17.9 0 0 1 12 21.75c-2.676 0-5.216-.584-7.499-1.632"/></svg> </div> <div class="sui-text-content"> <span class="sui-name">mock</span> <span class="sui-description">mock-admin</span> </div> </div> <script type="module" src="/mock/path/SSRUser.astro?astro&type=script&index=0&lang.ts"></script> "`;

exports[`Components Container tests > Shared components > ThemeManager component > render component 1`] = `
"<script>
    window.theme ??= (() => { 
        const defaultTheme = "system";
        const storageKey = "studiocms-theme-preference";
        const store =
            typeof localStorage !== "undefined"
            ? localStorage
            : { getItem: () => null, setItem: () => {} };
    
        const mediaMatcher = window.matchMedia("(prefers-color-scheme: light)");
        let systemTheme = mediaMatcher.matches ? "light" : "dark";
        mediaMatcher.addEventListener("change", (event) => {
            systemTheme = event.matches ? "light" : "dark";
            applyTheme(theme.getTheme());
        });
    
        function applyTheme(theme) {
            const resolvedTheme = theme === "system" ? systemTheme : theme;
            document.documentElement.dataset.theme = resolvedTheme;
            document.dispatchEvent(
            new CustomEvent("theme-changed", {
                detail: { theme, systemTheme, defaultTheme },
            })
            );
        }
    
        function setTheme(theme = defaultTheme) {
            store.setItem(storageKey, theme);
            applyTheme(theme);
        }
    
        function getTheme() {
            return store.getItem(storageKey) || defaultTheme;
        }
    
        function getSystemTheme() {
            return systemTheme;
        }
    
        function getDefaultTheme() {
            return defaultTheme;
        }
    
        return { setTheme, getTheme, getSystemTheme, getDefaultTheme };
    })();
    theme.setTheme(theme.getTheme());
</script> <script type="module" src="/mock/path/ThemeManager.astro?astro&type=script&index=0&lang.ts"></script>"
`;

exports[`Components Container tests > Shared components > TitleTags > render component 1`] = `"<title>Test Title</title><meta name="title" content="Test Title"><meta name="description" content="Test Description"><meta name="theme-color" content="#a581f3">"`;
